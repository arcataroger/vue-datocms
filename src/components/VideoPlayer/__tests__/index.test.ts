import { mount } from '@vue/test-utils';

require('matchmedia-polyfill');
require('matchmedia-polyfill/matchMedia.addListener');
(global as any).ResizeObserver = require('resize-observer-polyfill');

import { VideoPlayer } from '../';

describe('VideoPlayer', () => {
  beforeEach(() => {
    jest.spyOn(global.console, 'error').mockImplementation(() => {});
  });

  describe('with complete data', () => {
    const data = {
      muxPlaybackId: 'ip028MAXF026dU900bKiyNDttjonw7A1dFY',
      title: 'Title',
      width: 1080,
      height: 1920,
      blurUpThumb:
        'data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A',
    };

    it('generates a <mux-player/>', () => {
      const wrapper = mount(VideoPlayer, {
        propsData: { data },
      } as any);

      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
