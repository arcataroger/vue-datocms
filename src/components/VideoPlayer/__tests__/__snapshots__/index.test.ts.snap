// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer with complete data generates a <mux-player/> 1`] = `<mux-player stream-type="on-demand" title="Title" playback-id="ip028MAXF026dU900bKiyNDttjonw7A1dFY" placeholder="data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A" disable-cookies="true" disable-tracking="true" preload="metadata"></mux-player>`;
